xonsh/events.py

```diff
@@ -0,0 +1,169 @@
+"""
+Events for xonsh.
+
+In all likelihood, you want builtins.__xonsh_events__
+
+The best way to "declare" an event is something like::
+
+    __xonsh_events__.on_spam.doc("Comes with eggs")
+"""
+import abc
```

Comment:
Maybe we want to do `import collections.abc as cabc` or something.  Sorry about this @astronouth7303!